tools/ibc-test/src/tests/ibc/sudt_erc20_transfer.rs

```diff
@@ -0,0 +1,577 @@
+            .input(simple_input(st_cell))
+            .output(st_cell_output.clone())
+            .output_data(999u128.to_le_bytes().pack())
+            .cell_dep(sudt_cell_dep.clone())
+            .cell_dep(simple_dep(sudt_transfer_contract_out_point.clone()))
+            .build();
+        let st_cell_idx = tx.outputs().len() - 1;
+        let send_packet_tx = complete_tx_with_envelope(
+            &ckb_url,
+            &tx,
+            envelope,
+            ckb_sender_lock_script.clone(),
+            ckb_sender_key,
+        )
+        .unwrap();
+        let st_cell = packed::OutPoint::new_builder()
+            .tx_hash(send_packet_tx.hash())
+            .index(st_cell_idx.pack())
+            .build();
+        send_transaction(&ckb_url, send_packet_tx).unwrap();
+        log::info!("consumed ack");
+
+        // Receive sudt on ckb (write ack with forcerelay-ckb-sdk, transfer SUDT
+        // from st-lock to receiver)
+        log::info!("Receive SUDT");
+        let recv_packets = PacketCell::subscribe(sdk_client.clone(), sdk_config.clone())
+            .try_filter(|cell| futures::future::ready(cell.is_recv_packet()));
+        tokio::pin!(recv_packets);
+        let recv_packet = rt.block_on(recv_packets.try_next()).unwrap().unwrap();
+        let channel = rt
+            .block_on(IbcChannelCell::get_latest(&sdk_client, &sdk_config))
+            .unwrap();
+        let (tx, envelope) = assemble_write_ack_partial_transaction(
+            simple_dep(metadata_cell.out_point.into()),
+            simple_dep(channel_contract_cell.out_point.into()),
+            simple_dep(packet_contract_cell.out_point.into()),
+            &sdk_config,
+            channel,
+            recv_packet,
+            vec![1],
+        )
+        .unwrap();
+        let tx = tx
+            // st-cell input/output
+            .input(simple_input(st_cell))
+            .output(st_cell_output)
+            // 999 - 499 = 500.
+            .output_data(500u128.to_le_bytes().pack())
+            // received sudt output.
+            .output(
+                packed::CellOutput::new_builder()
+                    .lock(ckb_sender_lock_script.clone())
+                    .type_(Some(sudt_type_script.clone()).pack())
+                    .build_exact_capacity(Capacity::bytes(16).unwrap())
+                    .unwrap(),
+            )
+            .output_data(499u128.to_le_bytes().pack())
+            .cell_dep(sudt_cell_dep)
+            .cell_dep(simple_dep(sudt_transfer_contract_out_point.clone()))
+            .build();
+        let tx = complete_tx_with_envelope(
+            &ckb_url,
+            &tx,
+            envelope,
+            ckb_sender_lock_script.clone(),
+            ckb_sender_key,
+        )
+        .unwrap();
+        send_transaction(&ckb_url, tx).unwrap();
+        log::info!("Received SUDT");
+
+        // Sleep some time so the ack can be written to axon.
+        std::thread::sleep(Duration::from_secs(60));
```

Comment:
can here also run a Js script to listen and check the write-back of ack packet ?

Reply:
It's certainly possible. What function/event should be checked? Is it `AcknowledgePacket`?

Reply:
like using this [method](https://github.com/synapseweb3/ibc-solidity-contract/blob/master/contracts/core/05-port/IIBCModule.sol#L65) to emit a custom event which takes as a symbol of operation finished

Reply:
I think `AcknowledgePacket` is better because it doesn't require modifying the contract to emit an event just for testing. I'll try this.

Reply:
does it work, for listening `AcknowledgePacket` event from Axon?

Reply:
It seems like the `acknowledge_packet` function is never called on axon.